ta4j/src/main/java/org/ta4j/core/trading/rules/InSlopeRule.java

```diff
@@ -95,8 +95,9 @@ public boolean isSatisfied(int index, TradingRecord tradingRecord) {
 	Decimal val = diff.getValue(index);
 	boolean minSlopeSatisfied = minSlope.isNaN() ? true : val.isGreaterThanOrEqual(minSlope);
 	boolean maxSlopeSatisfied = maxSlope.isNaN() ? true : val.isLessThanOrEqual(maxSlope);
+	boolean isNaN = minSlope.isNaN() && maxSlope.isNaN();
```

Comment:
Dont know if this makes sense. As far as i understand the logic should be as follows:

- If `minSlope.isNaN()` ->  `cannot use this rule` >> `minSlopeSatisfied=false`
- If `maxSlope.isNaN()`  -> `do not consider upper bound` -> `maxSlopeSatisfied=true`

so it would be enough to change line 96 to:
`boolean minSlopeSatisfied = minSlope.isNaN() ? false: val.isGreaterThanOrEqual(minSlope);`

Reply:
No, it is correct. 

As shown in the constructor, both "minSlope" and "maxSlope" are optional values. However, at least one of those (or both, if both are given) must be a number. Imagine 3 cases: 

1) I want to test if maxSlope between a and b is 0.7, I do not care about minSlope. 
2) I want to test if minSlope between a and b is 0.2, I do not care about maxSlope.
3) I want to test if minSlope is 0.2 and maxSlope is 0.7, so both params but be a number.

Reply:
okay i see